compiler/next/include/chpl/uast/BuilderResult.h

```diff
@@ -120,8 +120,15 @@ class BuilderResult final {
     return errors_[i];
   }
 
-  // TODO: add something to iterate over the errors e.g.
-  // ErrorListIteratorPair errors() const
+  using ErrorGroup = std::vector<ErrorMessage>;
```

Comment:
It'd be nice to hide that the builder is even using a vector here. I think you could use that by making ErrorGroup an Iterable over vector<ErrorMessage>.

Reply:
Ok, so something like:

```cpp
using ErrorGroup = Iterable<std::vector<ErrorMessage>>;
using ErrorIterable = ErrorGroup;
```

?

Do you want me to also put `ErrorGroup` into private fields as well?

Reply:
I'm not following at the time why we need both of those but I'd expect we only need one to be public, at lest.

Reply:
I talked to Andrew and will just define `using ErrorIterable = Iterable<std::vector<thingy>>;` from now on. I think my original idea behind `ErrorGroup` was to make it private, but I'm not sure that works.